### PR TITLE
[batch] Create new job groups without ability to add jobs

### PR DESCRIPTION
**Pull request by jigold**
Stacked on #14016. This PR needs to have the client/server protocol for creating job groups for the four types of creation/update events hashed out and implemented. Basic tests are there. We still need tests for billing and cancellation to make sure the aggregation and cancellation operations work properly.

### CODE REVIEW

File: hail/python/hailtop/batch_client/aioclient.py [hunk 1]
```diff
@@ -952,45 +1190,97 @@ async def _submit_job_bunches(self,
...
+                                                                                   byte_job_group_specs_bunches[0] if n_job_group_bunches == 1 else [],
+                                                                                   job_group_bunch_sizes[0] if n_job_group_bunches == 1 else 0,
+                                                                                   job_progress_task,
+                                                                                   job_group_progress_task)
+                    else:
+                        start_job_group_id = None
+                        if len(self._job_groups) > 0:
+                            await self._submit_job_group_bunches(byte_job_group_specs_bunches, job_group_bunch_sizes, job_group_progress_task)
+                            # we need to recompute the specs now that the job group ID is absolute
+                            job_specs = [spec.to_dict() for spec in self._job_specs]
```
Comment (daniel-goldstein) [inline hail/python/hailtop/batch_client/aioclient.py:1280]: Is there some hidden mutation here that means running this code again produces a different result? That's pretty spooky and tells me that this should be restructured.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
This might not cover everything because I think some new commit came in while I was reviewing. The majority of lines changed here look great, I just have a couple core concerns with the API changes that I think are two strongly driven by the `*-fast` endpoints. In particular, I think `update-fast` is doing a lot here. `update-fast` is a convenience for bundling up the following steps:

1. Create an update *of jobs*
2. Submit jobs
3. Commit the update of those jobs

I might be wrong but it seems to me like that is now the de-facto way to also submit just a set of job groups with no jobs and that seems wrong. I think we would be better served by starting simpler and adding the ability to bundle operations once the right "primitives" are established. I think we would not be losing much if we just left all the `update` related operations alone and added a route for creating job groups and we would reduce the complexity greatly. We can then revisit by adding a way to "create a job group and a jobs update in one request"

**Comment by danking** [comment]
In the spirit of spreading out review burden a bit, I'm gonna pick up review for this PR. I'll be sure to read up on all the stacked conversations so I'm cache'd in before I review this one. I'll stick a review on here and you can dismiss when this is ready for a look :)

**Review by danking — CHANGES_REQUESTED** [review]
dismiss when ready for review

**Comment by jigold** [comment]
Closing in favor of #14170

**Pull request closed without merging**